### PR TITLE
Add PDF quick-editing to composition and selection assignments

### PR DESCRIPTION
**Pull request by nikolas**
* Add new pdf quick-edit annotator basics
* Add skeleton sherd pdf annotator
* Introduce isValidAnnotation and clean up some unnecessary special cases
* Use PDF iframe view in selection assignment

### CODE REVIEW

**Review by sdreher — COMMENTED** [review]
Okay - this looks fine to me. My one hesitation is around the naming conventions -- using "app2" and overloading the message names with "new" feels like something we'll need to undo further down the road. Can you think up a more specific/unique label we can use to represent this new sherdjs alternate path?

File: media/js/app/app2/index.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
+/* global Mustache: true */
```
[the remark this answers is not in the document]
Comment (nikolas) [reply media/js/app/app2/index.js]: I've renamed all the "app2" stuff to "quickeditor".

File: media/js/app/assetmgr/asset.js [hunk 2]
```diff
@@ -957,6 +963,8 @@
                 } else if (obj.start > obj.end) {
                     msg = 'The start time is greater than the end time.';
                 }
+            } else if (obj === 'missing pdfRect') {
+                msg = 'Select an annotation on the PDF.';
```
Comment (sdreher) [inline media/js/app/assetmgr/asset.js:967]: Prob better - "Select a rectangle on the PDF."

**Review by sdreher — COMMENTED** [review]
This is looking *really* good overall. I reviewed the code and also played around with the quickedit in composition, selection assignment and discussion assignment.

@todos would include comprehensive error handling and the cancel/clear behavior.

**Pull request merged**